go-update/src/main/java/org/reactome/release/goupdate/DuplicateCleaner.java

```diff
@@ -0,0 +1,215 @@
+		if (args.length > 0)
+		{
+			pathToResources = args[0];
+		}
+		try
+		{
+			Properties props = new Properties();
+			props.load(new FileInputStream(pathToResources));
+			DuplicateCleaner step = new DuplicateCleaner();
+			step.executeStep(props);
+		}
+		catch (Exception e)
+		{
+			e.printStackTrace();
+		}
+
+	}
+
+	@Override
+	public void executeStep(Properties props) throws Exception
+	{
+		this.adaptor = DuplicateCleaner.getMySQLAdaptorFromProperties(props);
+		int countOfDuplicatesAccessions = 0;
+		int instancesWithSignificantReferrers = 0;
+		
+		DuplicateReporter dupeReporter = new DuplicateReporter(adaptor);
+		Set<Long> dbIDsToDelete = new HashSet<Long>();
+
+		Map<String, Integer> duplicates = dupeReporter.getDuplicateAccessions();
+		countOfDuplicatesAccessions = duplicates.keySet().size();
+		// loop through all duplicated accessions.
+		for (String accession : duplicates.keySet())
+		{
+			logger.info("Accession {} is duplicated {} times.", accession, duplicates.get(accession));
+			List<Long> dbIDsWithNoReferrers = new ArrayList<Long>();
+			Map <Long,Integer> refCounts = dupeReporter.getReferrerCountForAccession(accession, goClasses);
+			// if there are referrers... refCounts should never be null, unless something catastrophic happens inside getReferrerCountForAccession and in that case, there will probably be a thrown exception.
+			if (refCounts.size() > 0)
+			{
+				// for each DB ID in referrer counts
+				for (Long dbId : refCounts.keySet())
+				{
+					// if there are referrers, we need to report them.
+					if (refCounts.get(dbId) > 0)
+					{
+						// update the counter of all instances with significant referrers.
+						instancesWithSignificantReferrers ++;
+						// log info about the referrers 
+						logReferrers(dbId, refCounts);
+					}
+					// if there are NO referrers, add the db_id to the list of DB IDs without referrers.
+					else
+					{
+						logger.debug("DB ID {} for accession {} has no significant referrers, and will be deleted.", dbId, accession);
+						dbIDsWithNoReferrers.add(dbId);
+					}
+				}
+			}
+			// update list of things to delete with IDs that have no referrers.
+			dbIDsToDelete.addAll(dbIDsWithNoReferrers);
+			// If no DB IDs have significant referrers, we should delete ALL DB_IDs for the accession, except for the newest one.
+			if (dbIDsWithNoReferrers.size() == refCounts.keySet().size())
+			{
+				GKInstance newestInstance = null;
+				// Need to try all classes since we don't know which class a GO Accession might be. Note: this.goClasses also contains "Compartment", but here "Compartment" is not used.
+				// It might be safe to use this.goClasses here, but since this bit of code did not use Compartment, we are not going to use this.goClasses here, for now.
+				for (String reactomeClass : Arrays.asList(ReactomeJavaConstants.GO_BiologicalProcess, ReactomeJavaConstants.GO_MolecularFunction, ReactomeJavaConstants.GO_CellularComponent))
```

Comment:
This list is used in many places in the code base.  It would consolidate and make the code more succinct to pull it into the GoUpdateConstants like:

```
static final List<String> GO_CLASSES = Arrays.asList(
    ReactomeJavaConstants.GO_BiologicalProcess,
    ReactomeJavaConstants.GO_MolecularFunction,
    ReactomeJavaConstants.GO_CellularComponent
);
```

Reply:
On closer inspection, this pattern (iterating over these 3 classes) only happens in two places - here and in `DuplicateReporter` (1 occurrence). `DuplicateCleaner` has `goClasses`, but as per the note, that also contains `Compartment`. In this particular case, `Compartment` is not used. So, I think it's OK to leave it as-is. `goClasses` takes care of the places where a longer list was used repeatedly.